src/hash_table.py

```diff
@@ -0,0 +1,31 @@
+# _*_ encoding: utf-8 _*_
+"""Simple implementation of a hash table."""
+
+
+class HashTable(object):
+    """Sample Python Hash Table."""
+
+    def __init__(self, slots):
+        """Init method for hash table."""
+        self.slots = slots
+        self._table = [[] for num in range(slots)]
+
+    def _hash(self, key):
+        """Render key string as integer with additive algorithm."""
+        if not isinstance(key, str):
+            raise TypeError('please give us a string')
```

Comment:
This is nitpicky, but `set` function should be what raises the TypeError. Granted, `_hash` is called when set is called, buuuuut yeah.